tensorflow/java/src/gen/cc/source_writer.h

```diff
@@ -75,18 +52,131 @@ class SourceWriter {
   // Indent(2)->Prefix("//") will result in prefixing lines with "  //".
   //
   // An empty value ("") will remove any line prefix that was previously set.
-  SourceWriter& Prefix(const char* line_prefix) {
-    line_prefix_ = line_prefix;
-    return *this;
+  SourceWriter& Prefix(const char* line_prefix);
+
+  // Writes a block of code or text.
+  //
+  // The data might potentially contain newline characters, therefore it will
+  // be scanned to ensure that each line is indented and prefixed properly,
+  // making it a bit slower than Append().
+  SourceWriter& Write(const string& text);
+
+  // Appends a piece of code or text.
+  //
+  // It is expected that no newline character is present in the data provided,
+  // otherwise Write() must be used.
+  SourceWriter& Append(const StringPiece& str);
```

Comment:
Any particular reason for `StringPiece` here but `string` in `Write`? Can they both be the same type?

Reply:
I remember that I *had* a reason but can't see what it is anymore, since `StringPiece` seems to offer everything that `SourceWriter.Write()` needs. I'll double-check and if it does, both signature will use that datatype. 

Personally, I prefer to stick to `StringPiece` better than `std::string` because if you append inlined code, like `source_writer.Append("return;")`, you avoid a useless 3-step conversion `char* -> std::string -> StringPiece`.

Reply:
Sounds good.